fal/controllers/anime_stats.py

```diff
@@ -16,8 +18,42 @@ def get_forum_posts(anime: Anime) -> int:
     '''
     Requests forum posts from Jikan, then sums up all the episode discussion
     thread replies
+
+    Currently uses regex to match for Episode Discussion. This is brittle and
+    will stop working once we get over 15 forum threads on an anime.
+
+    ASAP we should fix Jikan's API so that it can filter for episode discussion threads
+    in its search. Alternatively, we could remove the 15 search result limit but that
+    seems harder to do.
     '''
-    raise NotImplementedError()
+    week = config.getint("weekly info", "current-week")
+    jikan = jikanpy.Jikan()
+    forum_threads: List[Dict[str, Any]] = jikan.anime(
+        anime.id, extension='forum')['topics']
+
+    assert anime.name is not None
+    name = anime.name  # https://github.com/python/mypy/issues/4297
+
+    episode_discussions = list(filter(
+        lambda thread: re.fullmatch(
+            name + r' Episode \d{1,2} Discussion',
+            thread['title']
+        ),
+        forum_threads
+    ))
+
+    if len(episode_discussions) < week:
+        print(f"""
+            WARNING: did not find as many episode discussion threads for \
+                {anime.name} as the number of weeks we're in. Double check that \
+                    this is expected and manually update if necessary.
+            """)
+
+    return functools.reduce(
```

Comment:
I think we should use a generator expression as it seems easier to read.
```python
return sum(disc['replies'] for disc in episode_discussions)
```
We also wouldn't need the import of functools if we do it this way.

Reply:
done